### PR TITLE
Replace structopt with clap 3.1

### PR DESCRIPTION
**Pull request by bgilbert**
clap 3 drops `...` from the help text for arguments that can be repeated, and expects applications to make repeatability clear from the arg description: https://github.com/clap-rs/clap/issues/1571#issuecomment-541313239.  I'm not sure of a good way to do that, so we're currently leaving repeatable arguments implicit.

### CODE REVIEW

**Review by jlebon — APPROVED** [review]
Very nice!

**Comment by bgilbert** [comment]
CI correctly detected that clap 3.1 isn't in Fedora yet.  :tada:

**Comment by bgilbert** [comment]
Should be ready to go!

File: src/cmdline.rs [hunk 1]
```diff
@@ -315,10 +315,10 @@ pub struct InstallConfig {
     #[structopt(next_line_help(true))]
     pub network_dir: DefaultedString<NetworkDir>,
     /// Save partitions with this label glob
+    #[serde(skip_serializing_if = "is_default")]
     #[structopt(long, value_name = "lx")]
     // Allow argument multiple times, but one value each.  Allow "a,b" in
     // one argument.
-    #[serde(skip_serializing_if = "is_default")]
```
Comment (jlebon) [inline src/cmdline.rs:321]: OOC, was this actually causing compile issues?

File: src/cmdline.rs [hunk 2]
```diff
@@ -1072,7 +1072,11 @@ mod serializer {
             .context("reading subcommand help text")?;
 
         let mut serializer = Serializer {
-            help_text: String::from_utf8(help).context("decoding subcommand help text")?,
+            help_text: String::from_utf8(help)
+                .context("decoding subcommand help text")?
+                // add trailing space to each line, so push_field()
+                // option check works consistently for boolean flags
```
Comment (jlebon) [inline src/cmdline.rs:1084]: I think I understand this, but feels like the commit message could've elaborated a bit on what broke and why this fixes it.

**Review by jlebon — APPROVED** [review]
LGTM overall. Nice work!

File: src/cmdline.rs (hunk 1 above)
Comment (bgilbert) [reply src/cmdline.rs:321]: Not at all, it's just a cleanup.

File: src/cmdline.rs (hunk 2 above)
Comment (bgilbert) [reply src/cmdline.rs:1084]: Good call.  Updated.

**Review by jlebon — APPROVED** [review]

**Pull request merged**